src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -659,7 +660,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 {
                     countOfExplicitNames++;
 
-                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.UnderlyingMemberName(i))
+                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.TupleMemberName(i))
```

Comment:
The proposal is to disallow assigning name "Rest" to any item in the tuple, exactly for the purpose of disallowing shadowing of the "extension" field. Access tuple.Rest is still going to be allowed. What compatibility concerns do you have?